cloud_functions/data_processing/main.py

```diff
@@ -0,0 +1,343 @@
+        chunk_size=8192,
+        verbose=verbose,
+    )
+
+
+def download_mpatlas(
+    url=MPATLAS_URL,
+    bucket=BUCKET,
+    filename=MPATLAS_FILE_NAME,
+    archive_filename=ARCHIVE_MPATLAS_FILE_NAME,
+    verbose=True,
+):
+    if verbose:
+        print(f"downloading MPAtlas Zone Assessment from {url}")
+
+    r = requests.get(url)
+    r.raise_for_status()
+
+    print(f"saving MPAtlas Zone Assessment to gs://{bucket}/{archive_filename}")
+    # gdf = gpd.read_file(BytesIO(r.content))
+    # save_gdf_to_zipped_shapefile_gcs(gdf, filename, bucket, filename, verbose=True)
+    save_file_bucket(
+        r.content, r.headers.get("Content-Type"), archive_filename, bucket, verbose=verbose
+    )
+    duplicate_blob(bucket, archive_filename, filename, verbose=True)
+
+
+def download_protected_seas(
+    url=PROTECTED_SEAS_URL,
+    bucket=BUCKET,
+    filename=PROTECTED_SEAS_FILE_NAME,
+    archive_filename=ARCHIVE_PROTECTED_SEAS_FILE_NAME,
+    project=PROJECT,
+    verbose=True,
+):
+    """
+    Download Protected Seas data from website and saves zipfile to a current
+    file for calculations as well as an archive file
+    """
+    r = requests.get(url)
+    r.raise_for_status()
+
+    data = pd.DataFrame(r.json())
+
+    data["includes_multi_jurisdictional_areas"] = data["includes_multi_jurisdictional_areas"].map(
+        {"t": True, "f": False}
+    )
+
+    print(f"saving Protected Seas to gs://{bucket}/{archive_filename}")
+    upload_dataframe(bucket, data, archive_filename, project_id=project, verbose=verbose)
+    duplicate_blob(bucket, archive_filename, filename, verbose=True)
+
+
+def download_protected_planet_global(
+    current_filename, archive_filename, project_id=PROJECT, url=WDPA_GLOBAL_LEVEL_URL, bucket=BUCKET
+):
+    response = requests.get(url)
+    response.raise_for_status()
+    data = pd.read_csv(BytesIO(response.content))
+
+    upload_dataframe(bucket, data, archive_filename, project_id=project_id, verbose=True)
+    duplicate_blob(bucket, archive_filename, current_filename, verbose=True)
+
+
+def download_protected_planet_country(
+    current_filename,
+    archive_filename,
+    pp_api_key,
+    val="countries",
+    bucket=BUCKET,
+    project=PROJECT,
+    url=WDPA_API_URL,
+    per_page=50,
+    verbose=True,
+):
+    """
+    Download country level stats from Protected Planet API and saves to a current
+    file for calculations as well as an archive file
+    """
+    page = 1
+    all_areas = []
+    while True:
+        if verbose:
+            print(f"Fetching page {page}...")
+        params = {"token": pp_api_key, "per_page": per_page, "page": page}
+
+        response = requests.get(f"{url}/{val}", params=params)
+        response.raise_for_status()
+        data = response.json()
+
+        results = data.get(val, [])
+        if not results:
+            if verbose:
+                print("All pages fetched.")
+            break
+
+        all_areas.extend(results)
+        page += 1
```

Comment:
Slight preference for a more explicit `while` condition for readability:
```suggestion
  
  def fetch_data(params):
        response = requests.get(f"{url}/{val}", params=params)
        response.raise_for_status()
        data = response.json()
        return data.get(val, [])

   results = fetch_data(params)
    while results:
        if verbose:
            print(f"Fetching page {page}...")
        all_areas.extend(results)
        page += 1
        params = {"token": pp_api_key, "per_page": per_page, "page": page}
        results = fetch_data(params)
        
     if verbose:
       print("All pages fetched.")
```